p2p/tests/sync.rs

```diff
@@ -629,17 +650,46 @@ where
     );
 }
 
+// TODO: FIXME: Move somewhere.
+struct MakeP2pAddress {}
+
+impl MakeAddress for MakeP2pAddress {
```

Comment:
I'm still not convinced if this trait is needed to for the channel support but something like it would very convenient for making the tests less verbose. If the `NetworkService` would implement a `DefaultAddress` trait (or maybe just have a member function `NetworkService::default_address()`, there would be no need to call 'make_mock_addr()`/`make_libp2p_addr()` and the test code could instead initialize the managers like this:

```rust
    let (mut mgr1, mut conn1, _, mut pubsub, _) = make_sync_manager::<T>(T::default_address(), handle1).await;
    let (mut mgr2, mut conn2, _, _, _) = make_sync_manager::<T>(T::default_address(), handle2).await;
```

Reply:
This is actually specifically for the test, because there were methods you created to generate a test address. The trait should be called `MakeTestAddress`. The idea is that we want all addresses to be generated homogeneously so that we can use some generic method to run all the tests. Out of all the options I saw, I think this is the most reasonable one.

Reply:
I though about that, but in that case the `NetworkService` also needs to be implemented for `Libp2pServiceNetwork` and it doesn't seem right.

Originally I named this trait exactly MakeTestAddress, but then decided to shorten the name.  🙃  I'm going to rename it and move from the mocks to the tests.